### PR TITLE
Big Update

### PR DESCRIPTION
**Pull request by Sitryk**
- Added help display on certain errors
- Added `utils.py`, contains a pagify utility
- Added `checks.py`, contains a few checks used
- Rebuilt `config.py`, now you can load JSON and save easily
- Added a few Mod commands like, `kick` and `ban`
- Added some Owner commands, like `restart` and `eval`
- Ask for token on startup if no token saved yet
- Added error handling

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -1,26 +1,267 @@
...
+            for page in pages:
+                await ctx.send(page)
+        else:
+            pages = await self.formatter.format_help_for(ctx, ctx.command)
+            for page in pages:
+                await ctx.send(page)
+
+bot = PyBot(command_prefix='>>>', loop=loop)
+
+bot_defaults = {'token': None}
```
Comment (LaVieEstDure) [inline bot.py]: Theres already a defaults in the config file isnt there

File: bot.py [hunk 2]
```diff
@@ -1,26 +1,267 @@
...
+bot.settings = config.load_config(bot, 123, bot_defaults)
```
Comment (LaVieEstDure) [inline bot.py]: Put this and the command prefix settings in the `__init__`

File: bot.py [hunk 3]
```diff
@@ -1,26 +1,267 @@
...
+    elif isinstance(error, commands.DisabledCommand):
+        pass
+    elif isinstance(error, commands.CommandInvokeError):
+        err = ("\n--------------------\n"
+             "Error in Command: '{}'\n".format(ctx.command.qualified_name))
+        err += "".join(traceback.format_exception(type(error), error, error.__traceback__)) + "\n\n"
+        print(err)
+    elif isinstance(error, commands.CheckFailure):
+        await ctx.send("You can't use that command")
+    elif isinstance(error, commands.NoPrivateMessage):
```
Comment (LaVieEstDure) [inline bot.py]: I'm sure theres a better way than a huge if elif tree. Maybe with type(error), a dictionary somehow?

File: bot.py (hunk 1 above)
Comment (Sitryk) [reply bot.py]: there is a default of `{}` but if I default `'token'` in there, I don't have to check if `'token'` is in the json/dict and then test for None, on seconds thoughts I could just check if `'token'` is there and if not ask

File: checks.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import discord.utils
+from discord.ext import commands
+
+def bot_owner(ctx):
+    """
+    Check if the user is the bot owner
+    """
+    return ctx.author.id == 92861455442919424
```
Comment (LaVieEstDure) [inline checks.py]: no magic numbers pls

File: bot.py (hunk 3 above)
Comment (Sitryk) [reply bot.py]: sure I'll make a mapping

File: bot.py [hunk 5]
```diff
@@ -1,26 +1,267 @@
...
+        bot_py = os.path.realpath(__file__)
+        try:
+            ctx.send('Restarting...')
+        except:
+            pass
+        subprocess.Popen([python, "{}".format(bot_py)])
+        quit()
+
+    @commands.command()
+    async def eval(self, ctx, *, code):
```
Comment (LaVieEstDure) [inline bot.py]: ...I wouldnt make my own eval command. its probably better to use an online API. REPL.it shut down their free API I think but there was another one, I'll have to look

Comment (Sitryk) [reply bot.py]: I use it more for actually debugging the bot, not for testing code, that's why it's in the Owner only section, if we wanted to make a REPL or eval members could use then yeah I guess we should use an API

Comment (LaVieEstDure) [reply bot.py]: Ahh right, thats fine then

**Comment by Sitryk** [comment]
closing due to new PR

**Pull request closed without merging**